pkg/internal/tests/models/retry_test.go

```diff
@@ -108,3 +112,72 @@ func TestRetryWithNoSuccessStatusOnAnyRequest(t *testing.T) {
 		t.Errorf("Expected minimum time of %v, but got %v", expectedMinimumTime, elapsedTime)
 	}
 }
+
+// TestRetryWithNoSuccessStatusOnAnyRequest tests the retry mechanism with a server that always returns a 429 status code.
+func TestRetryWithNoSuccessStatusOnAnyRequest(t *testing.T) {
+	expectedStatusCode := http.StatusTooManyRequests
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(expectedStatusCode)
+	}))
+	defer server.Close()
+
+	var backoffTime = 2 * time.Second
+	var retryCount uint = 0
+	var expectedRetries uint = 3
+
+	sendRequest := func() (*http.Response, error) {
+		return http.Get(server.URL + "/notfound")
+	}
+
+	startTime := time.Now()
+
+	resp, err := wx.Retry(
+		sendRequest,
+		wx.NewRetryConfig(
+			wx.WithReturnHTTPStatusAsErr(false), // Use new behavior: only return actual network errors
+			wx.WithBackoff(backoffTime),
+			wx.WithOnRetryV2(func(n uint, resp *http.Response, err error) {
+				retryCount = n
+				if err != nil {
+					t.Errorf("In OnRetry, expected nil, got error: %v", err)
+				}
+
+				if resp == nil {
+					t.Errorf("In OnRetry, expected non-nil response, got nil")
+				}
+
+				if resp != nil && resp.StatusCode != expectedStatusCode {
+					t.Errorf("Expected status code %d, got %d", expectedStatusCode, resp.StatusCode)
+				}
+
+				log.Printf("Retrying request after response with status code: %d", resp.StatusCode)
+			}),
+		),
+	)
+
+	endTime := time.Now()
+
+	elapsedTime := endTime.Sub(startTime)
+	expectedMinimumTime := backoffTime * time.Duration(expectedRetries)
+
+	if err != nil {
```

Comment:
Note: The new behavior is that the error will be nil and the response will be non-nil.

This will give the user the opportunity to react to the response and use the response contents for troubleshooting purposes by logging/printing them to the end user.